patches/api/0496-API-to-allow-disallow-tick-sleeping.patch

```diff
@@ -0,0 +1,27 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Abel <abelvanhulst@gmail.com>
+Date: Tue, 12 Nov 2024 22:25:35 +0100
+Subject: [PATCH] API to allow/disallow tick sleeping
+
+
+diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
+index 0b78564256ebc647ebac402e549d86ab6e307c8d..f709429e258e487d75cf6fe3d2ee88d004890345 100644
+--- a/src/main/java/org/bukkit/Server.java
++++ b/src/main/java/org/bukkit/Server.java
+@@ -2572,4 +2572,16 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
+      */
+     boolean isOwnedByCurrentRegion(@NotNull Entity entity);
+     // Paper end - Folia region threading API
++
++    // Paper start - API to allow/disallow tick sleeping
++    /**
++     * Allow or disallow the server to sleep/pause.This only effects the next time the server is paused. If the server is currently paused, it will only go out of sleep when someone joins.
```

Comment:
>If the server is currently paused, it will only go out of sleep when someone joins

The diff  in tickServer should be moved up so that it stops sleep on the next tick

Reply:
What do you mean by this?

Reply:
There are earlier if statements you can edit that will be checked during tick sleep

Reply:
So the `if (j > 0)` statement?

Reply:
This is not resolved, see my last comment, also not sure why you have deleted comments

Reply:
> This is not resolved, see my last comment, also not sure why you have deleted comments

I had also posted the first message I deleted in the main thread, and the other one was just me messing around

Reply:
I have moved it down to the getPlayerCount() if-statement()